rasa/nlu/utils/pattern_utils.py

```diff
@@ -153,4 +154,15 @@ def extract_patterns(
             )
         )
 
+    # validate regexes, raise Error when invalid
+    for pattern in patterns:
+        try:
+            re.compile(pattern["pattern"])
```

Comment:
Compiling regex could be slowish and [Python recommends reusing regex objects](https://docs.python.org/3/library/re.html#re.compile). Would it work to catch the exception during later compilation while training and raise exception? I mean in regex_featurizer is compiling anyway, catch exception there and raise. Not sure if we lose pattern name etc. over there? Otherwise this PR is fine and the function here would validate albeit run a little slower.

Reply:
The regex_featurizer compiles when calling `re.finditer`. This method takes a few parameters which messes with the error returned.
I agree that compiling more than once is less efficient, but I feel like it is more logical to validate on the utility side.